### ISSUE TITLE
wp.tile_assign() fails when source is a register tile (e.g. from wp.tile_map())

### ISSUE THREAD

**Issue by shi-eric** [NVIDIA/warp#1232]
## Bug Description

`wp.tile_assign()` fails at CUDA compilation time when the source tile is a **register tile** — for example, the output of `wp.tile_map()`. There are two related issues in the C++ implementation (`tile.h`):

### 1. Forward pass: no template overload for register tile sources

The `tile_assign` implementation accesses source data via `src.data(c)`, which is the shared-memory tile access pattern. Register tiles (`tile_register_t`) use a different access pattern (`src.data[reg]`), so NVRTC reports:

```
error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    dest.data(c + offset) = src.data(c);
                            ^
```

### 2. Backward pass: `src.grad` doesn't exist on register tiles

The adjoint `adj_tile_assign` tries to accumulate gradients via `src.grad(c)`, but register tiles don't have a `.grad()` member:

```
error: class "tile_register_t<...>" has no member "grad"
    src.grad(c) += dest.grad(c + offset);
        ^
```

Additionally, even once the gradient accumulation compiles, `adj_tile_assign` doesn't zero the overwritten destination gradient entries. Since `tile_assign` overwrites destination values, those entries no longer contribute to any pre-assignment computation — failing to zero them causes double-counting during backpropagation.

## Minimal Repro

```python
import warp as wp
import numpy as np

TILE_M = 16
TILE_N = 32

@wp.func
def double_val(x: float):
    return x * 2.0

@wp.kernel
def repro_kernel(src: wp.array2d(dtype=float), dst: wp.array2d(dtype=float), col_idx: int):
    a = wp.tile_load(src, shape=(TILE_M, TILE_N), storage="shared")

    # Take a column view, apply tile_map (returns a register tile), assign back
    col_view = wp.tile_view(a, offset=(0, col_idx), shape=(TILE_M, 1))
    mapped = wp.tile_map(double_val, col_view)  # register tile
    wp.tile_assign(a, mapped, offset=(0, col_idx))  # fails here

    wp.tile_store(dst, a)

col_idx = 7
src_np = np.random.default_rng(42).random((TILE_M, TILE_N)).astype(np.float32)
src = wp.array(src_np, dtype=float)
dst = wp.zeros((TILE_M, TILE_N), dtype=float)

wp.launch_tiled(repro_kernel, dim=[1], inputs=[src, dst, col_idx], block_dim=32)

expected = src_np.copy()
expected[:, col_idx] *= 2.0
result = dst.numpy()
print("Match:", np.allclose(result, expected, atol=1e-6))
```

## Expected Behavior

The kernel should compile and run successfully. Column 7 of the source should be doubled in the output.

## Actual Behavior

NVRTC compilation error (`NVRTC_ERROR_COMPILATION`) with two errors in `tile.h` — one for the forward pass data access and one for the backward pass gradient access.

## Environment

- Warp 1.12.0.dev0 (also affects 1.11.x)
- CUDA Toolkit 13.1, Driver 13.1

**Comment by shi-eric** [comment]
Added in c9d9171b569f9a9c77712aba5206f3bb58033971
